### PR TITLE
fix some filters being dropped from history tables

### PR DESCRIPTION
**Pull request by JCOR11599**
Changes:
 - run `pruneTables` in `prePreparedSelector`, to transfer projections correctly
 - added `Filters()` getter into `sql.FilteredTables` interface 
 - `transferProjections` also copies over table filters

Test PR:
 - https://github.com/dolthub/dolt/pull/4163

### CODE REVIEW

File: enginetest/memory_engine_test.go [hunk 1]
```diff
@@ -118,52 +118,56 @@ func TestJSONTableQueries(t *testing.T) {
 
 // Convenience test for debugging a single query. Unskip and set to the desired query.
 func TestSingleQuery(t *testing.T) {
-	t.Skip()
+	//t.Skip()
```
Comment (max-hoffman) [inline enginetest/memory_engine_test.go:121]: reskip

File: enginetest/memory_engine_test.go [hunk 2]
```diff
@@ -118,52 +118,56 @@ func TestJSONTableQueries(t *testing.T) {
...
+	//engine.Analyzer.Debug = true
+	//engine.Analyzer.Verbose = true
 
 	enginetest.TestQueryWithEngine(t, harness, engine, test)
 }
 
 // Convenience test for debugging a single query. Unskip and set to the desired query.
 func TestSingleQueryPrepared(t *testing.T) {
-	t.Skip()
+	//t.Skip()
```
Comment (max-hoffman) [inline enginetest/memory_engine_test.go:149]:
```suggestion
	t.Skip()
```

File: enginetest/memory_engine_test.go (hunk 1 above)
Comment (max-hoffman) [reply enginetest/memory_engine_test.go:121]:
```suggestion
	t.Skip()
```

File: sql/analyzer/analyzer.go [hunk 3]
```diff
@@ -436,13 +445,16 @@ func postPrepareRuleSelector(id RuleId) bool {
...
+		//pruneTablesId,
 		resolveColumnsId,
 		resolveColumnDefaultsId,
 		expandStarsId,
+		//reorderProjectionId,
 
 		// OnceAfterDefault
 		pushdownFiltersId,
 		subqueryIndexesId,
+		//eraseProjectionId,
```
Comment (max-hoffman) [inline sql/analyzer/analyzer.go:457]: 1) all 3 of these are pre-only, cut from post?
2) delete commented lines if you're cutting

**Review by max-hoffman — APPROVED** [review]
LGTM, some cleanup nits, the PR summary is good, PR title could maybe allude to prepareds dropping FilteredTable filters or something more specific

File: sql/analyzer/analyzer.go (hunk 3 above)
Comment (JCOR11599) [reply sql/analyzer/analyzer.go]: 1. These are all things I thought needed to be done post, but I thought wrong.
2. ok

**Pull request merged**